VDA-StorageNode-Registry/test/index.test.ts

```diff
@@ -0,0 +1,939 @@
+                        const node = await contract.getNodeByAddress(users[i].address);
+                        checkGetNodeResult(node, storageNodes[i]);
+                    }
+                })
+            })
+
+            describe("Get by Endpoint", () => {
+                it("Failed: Unregistered endpoint", async () => {
+                    const unregisteredEndpoint = "https://unregistered"
+
+                    await expect(
+                        contract.getNodeByEndpoint(unregisteredEndpoint)
+                    ).to.be.revertedWithCustomError(contract, "InvalidEndpointUri");
+                })
+
+                it("Failed: Remove started", async () => {
+                    const currentSnapshot = await takeSnapshot();
+
+                    // Remove start
+                    const blockTime = await time.latest();
+                    const unregisterTime = blockTime + days(30);
+
+                    await checkRemoveNodeStart(users[0], unregisterTime);
+
+                    // Get
+                    await expect(
+                        contract.getNodeByEndpoint(storageNodes[0].endpointUri)
+                    ).to.be.revertedWithCustomError(contract, "InvalidEndpointUri")
+                    
+                    await currentSnapshot.restore();
+                })
+
+                it("Success", async () => {
+                    for (let i = 0; i < users.length; i++) {
+                        const node = await contract.getNodeByEndpoint(storageNodes[i].endpointUri);
+                        checkGetNodeResult(node, storageNodes[i]);
+                    }
+                })
+            })
+
+            describe("Get by Country", () => {
+                it("Return empty array for unregistered country", async () => {
+                    const unregistedCode = "SG";
+                    assert(storageNodes.findIndex(item => item.countryCode === unregistedCode) === -1);
+
+                    expect(await contract.getNodesByCountry(unregistedCode)).to.deep.equal([]);
+                })
+
+                it("Success", async () => {
+                    const allCountryCodes = storageNodes.map(item => item.countryCode);
+                    const countryCodes = [...new Set(allCountryCodes)]
+
+                    for (let i = 0; i < countryCodes.length; i++ ){
+                        const orgCountryNodes = storageNodes.filter(item => item.countryCode === countryCodes[i])
+
+                        const nodesReturned = await contract.getNodesByCountry(countryCodes[i]);
+
+                        expect(orgCountryNodes.length).to.equal(nodesReturned.length);
+
+                        for (let j = 0; j < orgCountryNodes.length; j++) {
+                            checkGetNodeResult(nodesReturned[j], orgCountryNodes[j]);
+                        }
+                    }
+                })
+            })
+
+            describe("Get by Region", () => {
```

Comment:
Is there a solidity limit to how many results this will return?

Reply:
Please check following:
- https://betterprogramming.pub/issues-of-returning-arrays-of-dynamic-size-in-solidity-smart-contracts-dd1e54424235
- https://github.com/openethereum/parity-ethereum/issues/6293